packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -141,13 +141,16 @@ class GoTrueClient {
   /// [password] is the password of the user
   ///
   /// [data] sets [User.userMetadata] without an extra call to [updateUser]
+  ///
+  /// [channel] Messaging channel to use (e.g. whatsapp or sms)
   Future<AuthResponse> signUp({
     String? email,
     String? phone,
     required String password,
     String? emailRedirectTo,
     Map<String, dynamic>? data,
     String? captchaToken,
+    OtpChannel? channel,
```

Comment:
I like to make things non-nullable as possible.

```suggestion
    OtpChannel channel = OtpChannel.sms,
```

This way it makes it clear for the dev the default value when hovered in the IDE, for example. This could be mentioned in the inline docs as well